pynest/nest/lib/hl_api_types.py

```diff
@@ -547,6 +547,11 @@ def __getattr__(self, attr):
         if not self:
             raise AttributeError("Cannot get attribute of empty NodeCollection")
 
+        # IPython looks up this method when doing pretty printing
+        # Without special casing this, this would go through SLI which is *slow*
+        if attr == "_ipython_canary_method_should_not_exist_":
+            raise KeyError(attr)
```

Comment:
How about 

```suggestion
            super().__getattribute__(attr)
```

as this will raise exactly the same AttributeError a normal object would. One could do the same for the numpy compatibility check right below.

Reply:
@Helveg Actually, this does not work. Some more experimentation showed that to stop IPython here, one needs to raise any kind of exceptions *except* an `AttributeError`. But `super.__getattribute__(attr)` wil raise exactly that and we are as far as before. I will change the exception to `NotImplementedError`,  which seems to make more sense here and add another comment, but then merge without further review if the tests pass.